code/inc/gfx/element/gfx_light.hpp

```diff
@@ -0,0 +1,152 @@
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+**************************************************************************************************/
+
+#pragma once
+
+/*=================================================================================================
+** 1.  REFERENCES
+**===============================================================================================*/
+
+/*=================================================================================================
+** 2.  INCLUDE FILES
+**===============================================================================================*/
+
+#include "gfx/element/gfx_element.hpp"
+
+/*=================================================================================================
+** 3.  DECLARATIONS
+**
+** 3.1 Macros
+**===============================================================================================*/
+
+/*=================================================================================================
+** 3.2 Types and Classes
+**===============================================================================================*/
+namespace GFX
+{
+    namespace Element
+    {
+        /**************************************************************************************************/
+        /**
+        * \par Details: 
+        */
+        class Light : public Element
+        {
+
+        public:
+            /*********************************/
+            // Public type definitions
+            /*********************************/
+
+            /*********************************/
+            // Public member variables
+            /*********************************/
+
+            glm::vec3 m_Position;
+            glm::vec3 m_Color;
+            float32_t m_Intensity;
+
+            /*********************************/
+            // Constructors/Destructor
+            /*********************************/
+
+            /**************************************************************************************************/
+            /**
+            * \brief The default constructor for the Light.
+            */
+            Light(void);
+
+            /**************************************************************************************************/
+            /**
+            * \brief The default destructor for the Light.
+            */
+            ~Light(void);
+
+            /*********************************/
+            // Public functions
+            /*********************************/
+
+            /**************************************************************************************************/
+            /**
+            * \brief 
+            * 
+            * \param[in] 
+            * 
+            * \retval 
+            */
```

Comment:
Fill in this function comment.

Reply:
Fixed https://github.com/tdiego1/worldweaver/pull/19/commits/1a0df967e70f1752177ebca99e4618e4ec06c26d